acme/acme.go

```diff
@@ -326,31 +353,48 @@ func (a *ACME) CreateLocalConfig(tlsConfig *tls.Config, checkOnDemandDomain func
 }
 
 func (a *ACME) getCertificate(clientHello *tls.ClientHelloInfo) (*tls.Certificate, error) {
+	var wildcardCertificate *tls.Certificate
 	domain := types.CanonicalDomain(clientHello.ServerName)
 	account := a.store.Get().(*Account)
-	//use regex to test for wildcard certs that might have been added into TLSConfig
+
+	// Check if we have a wildcard cert into TLSConfig that could be used
+	log.Debugf("Checking wildcard certificate matching for %v", domain)
 	for k := range a.TLSConfig.NameToCertificate {
 		selector := "^" + strings.Replace(k, "*.", "[^\\.]*\\.?", -1) + "$"
 		match, _ := regexp.MatchString(selector, domain)
 		if match {
-			return a.TLSConfig.NameToCertificate[k], nil
+			log.Debugf("Found a wildcard certificate to use as fallback: %v", k)
+			wildcardCertificate = a.TLSConfig.NameToCertificate[k]
```

Comment:
If this is the only time `wildcardCertificate` is used, why not just return it directly? Why store it, and check if it is empty later?

Seems like more code execution unnecessarily.